File: app/models/item.py
from sqlalchemy import Column, Integer, String, Text, Float, Boolean, DateTime, JSON
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.config.database import Base


class Item(Base):
    __tablename__ = "items"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    description = Column(Text)
    item_type = Column(String(50), nullable=False)  # weapon, armor, consumable, tool, etc.
    rarity = Column(String(20), default="common")  # common, uncommon, rare, epic, legendary
    
    # Estatísticas do item
    weight = Column(Float, default=0.0)
    value = Column(Integer, default=0)  # Valor em moedas de ouro
    quantity = Column(Integer, default=1)
    
    # Propriedades específicas
    is_magical = Column(Boolean, default=False)
    is_consumable = Column(Boolean, default=False)
    is_equippable = Column(Boolean, default=True)
    
    # Dados em JSON para propriedades específicas
    properties = Column(JSON, default=dict)  # Dano, AC, bônus, etc.
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # Relationships
    characters = relationship("Character", secondary="character_items", back_populates="items")

File: lib/models/item.dart
class Item {
  final String id;
  final String characterId;
  final String name;
  final String category; // armas, equipamento, consumível, etc.
  final double weight;
  final String? description;
  final int quantity;

  const Item({
    required this.id,
    required this.characterId,
    required this.name,
    required this.category,
    this.weight = 0,
    this.description,
    this.quantity = 1,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id'].toString(),
      characterId: json['character_id'].toString(),
      name: json['name'] as String? ?? '',
      category: json['category'] as String? ?? 'equipamento',
      weight: (json['weight'] as num?)?.toDouble() ?? 0.0,
      description: json['description'] as String?,
      quantity: (json['quantity'] as num?)?.toInt() ?? 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'category': category,
      'weight': weight,
      'quantity': quantity,
      if (description != null) 'description': description,
    };
  }
}
